discovery-infra/day2.py

```diff
@@ -0,0 +1,134 @@
+            )
+
+    set_cluster_pull_secret(client, cluster_id, args.pull_secret)
+
+    image_path = os.path.join(
+            consts.IMAGE_FOLDER,
+            f'{args.namespace}-installer-image.iso'
+        )
+    client.generate_and_download_image(
+            cluster_id=cluster.id,
+            image_path=image_path,
+            ssh_key=args.ssh_key,
+            )
+
+    day2_nodes_flow(client, internal_cluster_name, cluster, image_path, args.number_of_workers, args.api_vip_ip, args.api_vip_dnsname, args.namespace)
+
+
+def day2_nodes_flow(client, cluster_name, cluster, image_path, num_worker_nodes, api_vip_ip, api_vip_dnsname, namespace):
+    tf_network_name, total_num_nodes = apply_day2_tf_configuration(cluster_name, num_worker_nodes, api_vip_ip, api_vip_dnsname, namespace)
+    with utils.file_lock_context():
+        return utils.run_command(
+            f'make _apply_terraform CLUSTER_NAME={cluster_name}'
+        )
+    time.sleep(5)
+
+    utils.wait_till_nodes_are_ready(
+        nodes_count=total_num_nodes, network_name=tf_network_name
+    )
+
+    waiting.wait(
+        lambda: utils.are_libvirt_nodes_in_cluster_hosts(
+            client, cluster.id, num_worker_nodes
+        ),
+        timeout_seconds=consts.NODES_REGISTERED_TIMEOUT,
+        sleep_seconds=10,
+        waiting_for="Nodes to be registered in inventory service",
+    )
+
+    utils.wait_till_all_hosts_are_in_status(
+        client=client,
+        cluster_id=cluster.id,
+        nodes_count=num_worker_nodes,
+        statuses=[
+            consts.NodesStatus.KNOWN
+        ],
+        interval=30,
+    )
+
+    if args.install_cluster:
+        time.sleep(10)
+        client.install_day2_cluster(cluster.id)
+
+        utils.wait_till_all_hosts_are_in_status(
+            client=client,
+            cluster_id=cluster.id,
+            nodes_count=num_worker_nodes,
+            statuses=[
+                consts.NodesStatus.DAY2_INSTALLED
+            ],
+            interval=30,
+        )
+
+
+def apply_day2_tf_configuration(cluster_name, num_worker_nodes, api_vip_ip, api_vip_dnsname, namespace):
+    tf_folder = utils.get_tf_folder(cluster_name, namespace)
+    configure_terraform_workers_nodes(tf_folder, num_worker_nodes)
+    configure_terraform_api_dns(tf_folder, api_vip_ip, api_vip_dnsname)
+    return get_network_nodes_from_terraform(tf_folder)
+
+
+def configure_terraform_workers_nodes(tf_folder, num_worker_nodes):
+    tfvars = utils.get_tfvars(tf_folder)
+    num_workers = tfvars['worker_count']
+    num_workers = num_workers + num_worker_nodes
+    tfvars['worker_count'] = num_workers
+    set_workers_ips_by_type(tfvars, num_worker_nodes, 'libvirt_master_ips', 'libvirt_worker_ips')
+    set_workers_ips_by_type(tfvars, num_worker_nodes, 'libvirt_secondary_master_ips', 'libvirt_secondary_worker_ips')
+    utils.set_tfvars(tf_folder, tfvars)
+
+
+def configure_terraform_api_dns(tf_folder, api_vip_ip, api_vip_dnsname):
```

Comment:
asked previously but can't find the answer. Can't we add dns as part of tfvars?

Reply:
i would prefer not to: we don't need it for day1, and day2 is eventually a temp solution. I think it's better to update the main on the run